connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -152,8 +152,6 @@ public JsonNode toJson(final Schema schema, final Object value, final JsonConver
                 return switch (config.decimalFormat()) {
                     case NUMERIC -> JSON_NODE_FACTORY.numberNode(decimal);
                     case BASE64 -> JSON_NODE_FACTORY.binaryNode(Decimal.fromLogical(schema, decimal));
-                    default ->
-                        throw new DataException("Unexpected " + JsonConverterConfig.DECIMAL_FORMAT_CONFIG + ": " + config.decimalFormat());
```

Comment:
Is this change necessary? I think the error should still be thrown if the format is not expected.